http/src/client/mod.rs

```diff
@@ -1482,6 +1486,12 @@ impl Client {
             }
         }
 
+        if let Some(default_headers) = &self.state.default_headers {
+            for (name, value) in default_headers {
+                builder = builder.header(name, value);
+            }
+        }
```

Comment:
```suggestion
        if let (Some(default_headers), Some(headers)) = (&self.state.default_headers, builder.headers_mut()) {
            for (name, value) in default_headers {
                headers.insert(name, value);
            }
        }
```
We should overwrite existing headers instead of duplicating